src/Server/HTTPHandler.cpp

```diff
@@ -545,16 +545,27 @@ void HTTPHandler::processQuery(
                 auto header = current_output_format.getPort(IOutputFormat::PortKind::Main).getHeader();
                 used_output.exception_writer = [&, format_name, header, context_, format_settings](WriteBuffer & buf, int code, const String & message)
                 {
+                    if (used_output.out_holder->isCanceled())
+                    {
+                        chassert(buf.isCanceled());
+                        return;
+                    }
+
                     drainRequestIfNeeded(request, response);
                     used_output.out_holder->setExceptionCode(code);
+
                     auto output_format = FormatFactory::instance().getOutputFormat(format_name, buf, header, context_, format_settings);
                     output_format->setException(message);
                     output_format->finalize();
                     used_output.finalize();
+                    used_output.exception_is_written = true;
                 };
             }
             else
             {
+                if (used_output.out_holder->isCanceled())
```

Comment:
Should we add `chassert(buf.isCanceled());` here? Similar to the check above

Reply:
There is no var `buf` in that scope.